villain/formdps.c

```diff
@@ -0,0 +1,71 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+int main (int argc, char *argv[]) {
+	char * two = "2";
+	char * cspace = " ";
+        char * move = "mv ";
+	char * makerun2;
+        char * tmp;
+        char * mvrun2;
+        char * mmfiles;
+        char * mfiles;
+        FILE * fp;
+        int len;
+        if (strcmp(argv[1], "mv") == 0){
+           len = strlen(argv[2]);
+           if (!(tmp = malloc(len))) {
+              printf("malloc error");
+           }
+           len--;
+           strncpy(tmp, argv[2], len);
+           tmp[len] = '\0';
+           if (!(mvrun2 = malloc (2 * len + 6))){
+              printf("malloc error");
+           }
+           strcpy(mvrun2, argv[1]);
```

Comment:
I think `snprintf` can make the code here much easier to read.

Reply:
Yes, snprintf is a better option.